pkg/psmdb/backup/pbm.go

```diff
@@ -672,8 +831,78 @@ func (b *pbmC) GetRestoreMeta(ctx context.Context, name string) (*restore.Restor
 	return restore.GetRestoreMeta(ctx, b.Client, name)
 }
 
-func (b *pbmC) ResyncStorage(ctx context.Context, stg *config.StorageConf) error {
-	return resync.Resync(ctx, b.Client, stg, "")
+func (b *pbmC) ResyncMainStorage(ctx context.Context) error {
+	return b.SendCmd(ctx, ctrl.Cmd{Cmd: ctrl.CmdResync})
+}
+
+func (b *pbmC) ResyncMainStorageAndWait(ctx context.Context) error {
+	if err := b.ResyncMainStorage(ctx); err != nil {
+		return errors.Wrap(err, "start resync")
+	}
+
+	if err := b.WaitForResync(ctx); err != nil {
+		return errors.Wrap(err, "wait for resync")
+	}
+
+	return nil
+}
+
+func (b *pbmC) WaitForResync(ctx context.Context) error {
+	log := logf.FromContext(ctx)
+
+	startCtx, startCancel := context.WithTimeout(ctx, 30*time.Second)
+	defer startCancel()
+
+	ticker := time.NewTicker(50 * time.Millisecond)
+	defer ticker.Stop()
+
+	log.Info("waiting for resync to start")
+start:
+	for {
+		select {
+		case <-startCtx.Done():
+			return errors.New("resync is not started until deadline")
+		case <-ticker.C:
+			resyncRunning, err := b.HasLocks(startCtx, IsResync)
+			if err != nil {
+				return errors.Wrap(err, "check PBM locks")
+			}
+
+			if resyncRunning {
+				break start
```

Comment:
Why don't we simply break here and we have this goto-like logic? Do we need it?

Reply:
Rightright, we can do this which is more clear: https://goplay.tools/snippet/LMrk7nVCs0_n